plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployJob.java

```diff
@@ -164,8 +165,13 @@ private IStatus deployProject(Path credentialFile, IPath stagingDirectory, IProg
       optionalConfigurationFilesDirectory = stager.getOptionalConfigurationFilesDirectory();
     }
 
-    return deployer.deploy(stagingDirectory, credentialFile, deployPreferences,
-        optionalConfigurationFilesDirectory, stdoutOutputStream, monitor);
+    try {
+      return deployer.deploy(stagingDirectory, credentialFile, deployPreferences,
```

Comment:
So, I have been comparing

- `deployer.deploy()` throwing `CloudSdkNotFoundException` and catching it here in `DeployJob.deployProject()` to return an error `IStatus`, which is the current PR.

vs.

- `deploy.deploy()` consuming `CloudSdkNotFoundException` there inside and returning an error `IStatus`. (`DeployJob.deployProject()` code here won't change at all.)

I prefer the latter, because from the perspective of this method (`deployProject()`), there is no `CloudSdk` involved, which is like an implementation detail. Basically, `CloudSdk` is invisible from `DeployJob`. What do you think?

Reply:
done